docs/docs/hydro/live-collections/streams.md

```diff
@@ -3,10 +3,119 @@ sidebar_position: 2
+
+To track this behavior, stream have an `Order` type parameter that indicates whether the elements in the stream will have a deterministic order ([`TotalOrder`](pathname:///rustdoc/hydro_lang/stream/struct.TotalOrder)) or not ([`NoOrder`](pathname:///rustdoc/hydro_lang/stream/struct.NoOrder)). When the type parameter is omitted, it defaults to `TotalOrder` for brevity.
+
+If we send a stream from a cluster to a process, the return type will be a stream with `NoOrder`:
+
+```rust,no_run
+# use hydro_lang::*;
+# let flow = FlowBuilder::new();
+let workers: Cluster<()> = flow.cluster::<()>();
+let numbers: Stream<_, Cluster<_>, _, TotalOrder> =
+    workers.source_iter(q!(vec![1, 2, 3]));
+let process: Process<()> = flow.process::<()>();
+let on_p2: Stream<_, Process<_>, _, NoOrder> =
+    numbers.send_bincode(&process);
+```
+
+The ordering of a stream determines which APIs are available on it. For example, `map` and `filter` are available on all streams, but `last` is only available on streams with `TotalOrder`. This ensures that even when the network introduces non-determinism, the program will not compile if it tries to use an API that requires a deterministic order.
+
+A particularly common API that faces this restriction is [`fold`](pathname:///rustdoc/hydro_lang/stream/struct.Stream#method.fold) (and [`reduce`](pathname:///rustdoc/hydro_lang/stream/struct.Stream#method.reduce)). These APIs require the stream to have a deterministic order, since the aggregation may depend on the order of elements. For example, the following code will not compile because `fold` is not available on `NoOrder` streams. The error is a bit misleading, but the key part is that `fold` is not available on `NoOrder` streams:
+
+```compile_fail
+# use hydro_lang::*;
+# let flow = FlowBuilder::new();
+let workers: Cluster<()> = flow.cluster::<()>();
+let process: Process<()> = flow.process::<()>();
+let all_words: Stream<_, Process<_>, _, NoOrder> = workers
+    .source_iter(q!(vec!["hello", "world"]))
+    .map(q!(|x| x.to_string()))
+    .send_bincode_interleaved(&process);
+
+let words_concat = all_words
+    .fold(q!(|| "".to_string()), q!(|acc, x| acc += x));
+//   ^^^^ `hydro_lang::Stream<&str, hydro_lang::Process<'_>, hydro_lang::Unbounded, NoOrder>` is not an iterator
+```
+
+:::tip
+
+We use `send_bincode_interleaved` here to drop the cluster IDs which are included in `send_bincode`. See [Clusters](../locations/clusters.md) for more details.
+
+You'll notice that we aggregated an **asynchronously** updated stream, so the result is a `Singleton` live collection. For more details on the semantics of singletons, including how they are updated when new inputs arrive, see [Singletons and Optionals](./singletons-optionals.md).
```

Comment:
I think this needs a gentler discussion; this is opaque as written.  How about something like:
"Running an aggregation (`fold_*`, `reduce_*`) converts a `Stream` into a `Singleton`, as we see in the type signature here. The `Singleton` type is still "live" in the sense of a [`live collection`](ref), so updates to the `Stream` input cause updates to the `Singleton` output. See [Singletons and Optionals](./singletons-optionals.md) for more information.